internal/pkg/db/db.go

```diff
@@ -39,17 +41,22 @@ type (
 	Option func(*Backend)
 )
 
-func NewBackend(opts ...Option) *Backend {
+func NewBackend(opts *options.Options) (*Backend, error) {
 	backend := &Backend{
 		Backend: ent.NewBackend(),
 		Logger:  utils.NewLogger("db"),
 	}
 
-	for _, opt := range opts {
-		opt(backend)
+	backend.
+		Debug(opts.Debug).
+		WithDatabaseFile(filepath.Join(opts.CacheDir, DatabaseFile)).
+		WithLogger(opts.Logger)
+
+	if err := backend.InitClient(); err != nil {
+		return nil, fmt.Errorf("failed to initialize backend client: %w", err)
```

Comment:
Also, we should probably just remove the receiver methods altogether, since

- `NewBackend()` now potentially returns an error, so they can't be chained anymore
- Their usage isn't as clear outside the context of creating a new backend
- Since `backend.InitClient()` is now being called inside `NewBackend()`, those options have to be set beforehand, so setting them after the fact with the builder pattern/receiver methods would have unintended consequences

Just need to update the functional options to modify a `Backend` directly, e.g.

```go
// WithDatabaseFile sets the database file for the backend.
func WithDatabaseFile(file string) Option {
	return func(backend *Backend) {
		// Change this
		// backend.WithDatabaseFile(file)

		// to this
		backend.Backend.Options.DatabaseFile = file
	}
}
```